### PR TITLE
DEV-54: Add support for application config

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19555/HighFidelity-Beta-PR15594-d9f164b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19555/HighFidelity-Beta-Interface-PR15594-d9f164b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19555/HighFidelity-Beta-PR15594-d9f164b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19555/HighFidelity-Quest-Beta-PR15594-d9f164b.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19555/HighFidelity-Beta-PR15594-d9f164b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19555/HighFidelity-Beta-Interface-PR15594-d9f164b.exe

File: interface/src/main.cpp [hunk 1]
```diff
@@ -106,6 +108,49 @@ int main(int argc, const char* argv[]) {
...
+            auto contents = configFile.readAll();
+            QJsonParseError error;
+
+            auto doc = QJsonDocument::fromJson(contents, &error);
+            if (error.error) {
+                qWarning() << "Found application config, but could not parse it: " << error.errorString();
+            } else {
+                static const QString LAUNCHER_PATH_KEY = "launcherPath";
+                QString launcherPath = doc.object()[LAUNCHER_PATH_KEY].toString();
+                if (!launcherPath.isEmpty()) {
```
Comment (luiscuenca) [inline interface/src/main.cpp]: What if the path is invalid?

Comment (luiscuenca) [reply interface/src/main.cpp]: We decided that the current behavior is the right one for the moment.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19566/HighFidelity-Beta-PR15594-d9f164b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19566/HighFidelity-Beta-Interface-PR15594-d9f164b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19566/HighFidelity-Beta-PR15594-d9f164b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19566/HighFidelity-Quest-Beta-PR15594-d9f164b.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19566/HighFidelity-Beta-PR15594-d9f164b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19566/HighFidelity-Beta-Interface-PR15594-d9f164b.exe

**Pull request merged**